core/rawdb/accessors_chain.go

```diff
@@ -142,14 +142,16 @@ func ReadAllCanonicalHashes(db ethdb.Iteratee, from uint64, to uint64, limit int
 	return numbers, hashes
 }
 
+const EmptyNumber = ^uint64(0)
+
 // ReadHeaderNumber returns the header number assigned to a hash.
-func ReadHeaderNumber(db ethdb.KeyValueReader, hash common.Hash) *uint64 {
+func ReadHeaderNumber(db ethdb.KeyValueReader, hash common.Hash) uint64 {
```

Comment:
why can't we just make this return (uint64, bool) to signify that the uint64 returned is valid or not?

Reply:
Yeah, thats a way better idea